### ISSUE TITLE
log-likelihood of order/connectivity-agnostic training

### ISSUE THREAD

**Issue by wangleiphy** [karpathy/pytorch-made#1]
[Here](https://github.com/karpathy/pytorch-made/blob/master/run.py#L39) you simply average over the logits. However, did I understand correctly that the order/connectivity-agnostic training amounts to a mixture model with $p(x) = \sum_{o} p(x, o)$ ? If so, it appears that one needs to compute log-likelihood for each mask and then perform `logsumexp` to obtain the mixture log-likelihood.
